domain/schema/controller/sql/0014-objectstore-metadata.sql

```diff
@@ -1,26 +1,14 @@
-CREATE TABLE object_store_metadata_hash_type (
-    id INT PRIMARY KEY,
-    hash_type TEXT NOT NULL
-);
-
-CREATE UNIQUE INDEX idx_object_store_metadata_hash_type_name
-ON object_store_metadata_hash_type (hash_type);
-
-INSERT INTO object_store_metadata_hash_type VALUES
-(0, 'none'),
-(1, 'sha512-384');
-
 CREATE TABLE object_store_metadata (
     uuid TEXT NOT NULL PRIMARY KEY,
-    hash_type_id INT NOT NULL,
-    hash TEXT NOT NULL,
-    size INT NOT NULL,
-    CONSTRAINT fk_object_store_metadata_hash_type
-    FOREIGN KEY (hash_type_id)
-    REFERENCES object_store_metadata_hash_type (id)
+    sha_256 TEXT NOT NULL,
+    sha_384 TEXT NOT NULL,
+    size INT NOT NULL
 );
 
-CREATE UNIQUE INDEX idx_object_store_metadata_hash ON object_store_metadata (hash);
+-- Add a unique index for each hash and a composite unique index for both hashes
+-- to ensure that the same hash is not stored multiple times.
+CREATE UNIQUE INDEX idx_object_store_metadata_sha_256 ON object_store_metadata (sha_256);
+CREATE UNIQUE INDEX idx_object_store_metadata_sha_384 ON object_store_metadata (sha_384);
```

Comment:
I think it's more expensive to create this index than what it brings to the table, if we have collisions we should play the lottery xD